### ISSUE TITLE
Move device to it's own module (device.py)

### ISSUE THREAD

**Issue by yandy-r** [jedelman8/cpal#24]
What do you guys think about moving device() out of main.py and putting it in it's own module? This way we can have a true representation of the Model/View/Controller. Can also help in modularizing things a bit more.

So in this case:

Things contained under vendors are the **Model** for this.
Device would be the **Controller** 
and main would be the **view**, the representation of the data that Device is interpreting from the model classes.

Keeping device in main, can make it cumbersome as this grows.

Just an idea. Thoughts?

**Comment by wdennis** [comment]
Is there a overall design doc available? (If not, maybe a good idea?)

**Comment by wdennis** [comment]
This is why I ask, by the way ;)  http://t.co/9P6iNNXKU4

**Comment by yandy-r** [comment]
ha! that was freaking funny, stealing that picture

**Comment by jedelman8** [comment]
No, there isn't. But I do want to start one, or at lease start documenting better. 

Stay tuned.

**Comment by sjiveson** [comment]
I'm quite happy to help with this too if need be.
Cheers
Steve Iveson

Date: Sun, 6 Apr 2014 10:47:35 -0700

**Comment by yandy-r** [comment]
likewise

---

Yandy Ramirez
